source/github-helpers/github-url.ts

```diff
@@ -37,17 +37,18 @@ export default class GitHubURL {
 		ambiguousReference: string[]
 	): {branch: string; filePath: string} {
 		const branch = ambiguousReference[0];
+		const currentBranch = getCurrentCommittish();
 		// History pages might use search parameters
+		const branchFromSearch = this.searchParams.get('branch');
 		const filePathFromSearch = this.searchParams.getAll('path[]').join('/');
 		if (filePathFromSearch) {
+			this.searchParams.delete('branch');
 			this.searchParams.delete('path[]');
-			return {branch, filePath: filePathFromSearch};
+			return {branch: branchFromSearch ?? currentBranch ?? branch, filePath: filePathFromSearch};
```

Comment:
I don’t think `currentBranch` should be part of this at all. GitHubURL can receive a URL to any page so we can't just change it to use the current branch.